validator/accounts/account.go

```diff
@@ -222,12 +222,12 @@ func HandleEmptyKeystoreFlags(cliCtx *cli.Context, confirmPassword bool) (string
 }
 
 // Merge merges data from validator databases in sourceDirectories into a new store, which is created in targetDirectory.
-func Merge(ctx context.Context, sourceDirectories []string, targetDirectory string) error {
+func Merge(ctx context.Context, sourceDirectories []string, targetDirectory string) (err error) {
 	var sourceStores []*db.Store
 	defer func() {
 		for _, store := range sourceStores {
-			if err := store.Close(); err != nil {
-				err = errors.Wrapf(err, "Failed to close the database in %s", store.DatabasePath())
+			if deferErr := store.Close(); deferErr != nil {
+				err = errors.Wrapf(deferErr, "Failed to close the database in %s", store.DatabasePath())
```

Comment:
> So in this case, since you are deferring it. There is no where to return the error to, you should instead log it.

There is a way to return the error - by using a named return parameter (`err` in this case). The error inside `defer` needs to have another name, otherwise the `err = (...)` statement will assign the value to the variable whose scope is the `if` statement, not the function's return parameter.

I prepared an example for you:
https://play.golang.org/p/F1tTLk028CY

Reply:
Looks good to me